File: NestJSMySql/src/controllers/person.controller.ts
import {
  Body,
  Controller,
  Delete,
  Get,
  NotFoundException,
  Param,
  Patch,
  Post,
} from "@nestjs/common";
import { Person } from "src/entities/person.entity";
import { PersonService } from "src/services/person.service";

@Controller("/person")
export class PersonController {
  constructor(private PersonService: PersonService) {}

  @Get("")
  fetchAll() {
    return this.PersonService.fetchAll();
  }

  @Get("/:id")
  async fetchOne(@Param("id") id: string) {
    const Person = await this.PersonService.fetchOne(+id);

    if (!Person) throw new NotFoundException("Person not found");

    return Person;
  }

  @Post()
  create(@Body() Person: Person) {
    return this.PersonService.create(Person);
  }

  @Patch("/:id")
  async update(@Param("id") id: string, @Body() Person: Partial<Person>) {
    const receivedPerson = await this.PersonService.update(+id, Person);

    if (!receivedPerson) throw new NotFoundException("Person not found");

    return receivedPerson;
  }

  @Delete("/:id")
  async delete(@Param("id") id: string) {
    const receivedPerson = await this.PersonService.delete(+id);

    if (!receivedPerson) throw new NotFoundException("Person not found");

    return receivedPerson;
  }
}

File: SpringBootwithMongoDB/src/main/java/com/SpringBootwithMongoDB/controller/PersonController.java
package com.SpringBootwithMongoDB.controller;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

import com.SpringBootwithMongoDB.entity.Person;
import com.SpringBootwithMongoDB.service.PersonService;

@RestController
public class PersonController {

	@Autowired
	private PersonService personService;

	@GetMapping("/person")
	public List<Person> fetchAll() {
		return this.personService.fetchAll();
	}

	@GetMapping("/person/{id}")
	public Person fetchById(@PathVariable String id) {
		return this.personService.fetchById(id);
	}

	@DeleteMapping("/person/{id}")
	public void delete(@PathVariable String id) {
		this.personService.delete(id);
	}

	@PostMapping("/person")
	public Person create(@RequestBody Person person) {
		return this.personService.create(person);
	}
	
	@PutMapping("/person/{id}")
	public ResponseEntity<Object> update(@RequestBody Person person, @PathVariable String id) {
		return this.personService.update(person, id);
	}
}